rct229/rulesets/ashrae9012019/section19/section19rule4.py

```diff
@@ -0,0 +1,277 @@
+                    index_rmr="baseline",
+                    list_path="$.zones[*]",
+                )
+
+            def create_data(self, context, data):
+                building_segment_b = context.baseline
+
+                lighting_bldg_type_b = building_segment_b.get(
+                    "lighting_building_area_type"
+                )
+
+                is_lighting_bldg_area_defined = False
+                is_building_area_MF_dormitory_or_hotel = False
+                if not lighting_bldg_type_b:
+                    is_lighting_bldg_area_defined = True
+                elif lighting_bldg_type_b in [
+                    LIGHTING_BUILDING_AREA.DORMITORY,
+                    LIGHTING_BUILDING_AREA.HOTEL_MOTEL,
+                    LIGHTING_BUILDING_AREA.MULTIFAMILY,
+                ]:
+                    is_building_area_MF_dormitory_or_hotel = True
+
+                return {
+                    "is_lighting_bldg_area_defined": is_lighting_bldg_area_defined,
+                    "is_building_area_MF_dormitory_or_hotel": is_building_area_MF_dormitory_or_hotel,
+                }
+
+            class ZoneRule(RuleDefinitionListIndexedBase):
+                def __init__(self):
+                    super(
+                        Section19Rule4.RuleSetModelInstanceRule.BuildingSegmentRule.ZoneRule,
+                        self,
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, False),
+                        each_rule=Section19Rule4.RuleSetModelInstanceRule.BuildingSegmentRule.ZoneRule.SpaceRule(),
+                        index_rmr="baseline",
+                        list_path="$.spaces[*]",
+                        required_fields={
+                            "$": ["infiltration"],
+                        },
+                    )
+
+                def create_data(self, context, data):
+                    day_of_week_for_january_1 = data["day_of_week_for_january_1"]
+                    zone_b = context.baseline
+
+                    # check infiltration
+                    inf_pass_cooling = True
+                    multiplier_schedule_infiltration = zone_b["infiltration"][
+                        "multiplier_schedule"
+                    ]
+                    multiplier_sch = data[multiplier_schedule_infiltration]
+                    hourly_values = multiplier_sch["hourly_values"]
+
+                    most_used_weekday_hourly_schedule = (
+                        get_most_used_weekday_hourly_schedule(
+                            hourly_values, day_of_week_for_january_1
+                        )
+                    )
+                    for x in range(0, int(len(hourly_values) / 24), 24):
+                        if (
+                            most_used_weekday_hourly_schedule
+                            != hourly_values[x : x + 24]
+                        ):
+                            inf_pass_cooling = False
+                            break
```

Comment:
This logic is very confusing. I do not think this is correct in RDS. I will post an issue.

Reply:
Follow up the issue: https://github.com/pnnl/ruleset-checking-tool/issues/1020
Please note there is a data that is currently not in the schema yet.
A good temporary way to do it is adding this data to RCT schema but I would suggest to wait until we are clear about the data (name, type etc.)

Reply:
Thank you Weili. Addressed.